src/net_processing.cpp

```diff
@@ -4564,7 +4574,7 @@ void PeerManagerImpl::ProcessMessage(
 
         // Bypass the normal CBlock deserialization, as we don't want to risk deserializing 2000 full blocks.
         unsigned int nCount = ReadCompactSize(vRecv);
-        if (nCount > MAX_HEADERS_RESULTS) {
+        if (nCount > GetHeadersLimit(pfrom, msg_type == NetMsgType::HEADERS2)) {
```

Comment:
nit: can commit result to variable and reuse it further down, also consider changing misbehavior text to mention headers2 when appropriate due to different limit.

```suggestion
        const bool is_compressed{msg_type == NetMsgType::HEADERS2};
        if (nCount > GetHeadersLimit(pfrom, is_compressed)) {
```